apps/game_backend/lib/game_backend/users/currencies.ex

```diff
@@ -155,4 +155,20 @@ defmodule GameBackend.Users.Currencies do
       {:error, "failed"}
     end
   end
+
+  @doc """
+  Gets how much a user has of a given currency by its name.
+  """
+  def get_amount_of_currency_by_name!(user_id, currency_name),
+    do:
+      user_id
+      |> get_amount_of_currency(get_currency_by_name!(currency_name).id)
```

Comment:
We're doing two queries here instead of one why not doing it all at once?
```
get_amount_of_currency_by_name!(currency_name)

def get_amount_of_currency_by_name(currency_name) do
Repo.one(
        from(uc in UserCurrency,
          join: currency in assoc(uc, :currency),
          where: uc.user_id == ^user_id and currency.name == ^currency_name,
          select: uc.amount
        )
      ) || 0
end
```

Do you think this is better?

Reply:
Yes it's definitely better!